server/main.go

```diff
@@ -80,6 +85,18 @@ type Main struct {
 	TreeDeleteMinInterval time.Duration
 }
 
+func (m *Main) healthz(rw http.ResponseWriter, req *http.Request) {
+	if m.IsHealthy != nil {
+		if err := m.IsHealthy(req.Context()); err != nil {
```

Comment:
I think (could be wrong) that the request context defaults to context.Background() so might need an explicit deadline here.

Reply:
Done